### DISCUSSION TITLE
question: how to make a plugin switched on by default (in settings.yml)

### DISCUSSION THREAD

**Discussion by georg-mayer** [searx/searx#2503] [Development]
Hello,

what do in settings.yml I have to do to turn a plugin (e.g. Infinite Scroll) on by default (for all users)? 

I tried with the same syntax as for engines, i.e. 

```
- plugins:
      name: infinite scroll
      disabled: False
```

but this only made searx fail to start.

Thanks for your help! Cheers,
Georg

**Comment by rasos** [comment]
Yes I also would like to offer infinite scrolling as default for any user. This did not really get me further http://searx.github.io/searx/dev/plugins.html

So I patched the file       

      /searx/searx/plugins/infinite_scroll.py

with        

      default_on = True

**Comment by 0xhtml** [comment]
You can select which plugins are on by default by specifying `enabled_plugins` in `settings.yml`.

Example:
```yaml
enabled_plugins:
    - "HTTPS rewrite"
    - "Infinite scroll"
```

**Reply by ESMP** [reply]
Whenever I uncomment those lines, searx hangs with a 504 Gateway Time-out. What am I doing wrong when enabling plugins in settings.yml? If I comment the plugins' lines again, searx loads just fine.

**Reply by unixfox** [reply]
Check the logs or switch to searxng which have working plugins.

**Reply by ESMP** [reply]
I think I found the problem... indentation in the enabled_plugins line.
Either way, I switched to NG. Thanks for the suggestion!
